File: app/src/main/java/com/example/safedrive/nav_components/dashborad/Dashborad.java
// Dashborad.java
package com.example.safedrive.nav_components.dashborad;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.View.OnClickListener;
import android.widget.ImageView;
import androidx.fragment.app.Fragment;
import com.example.safedrive.R.id;
import com.example.safedrive.activities.Prediction;
import com.example.safedrive.activities.ViewData;
import com.example.safedrive.activities.cnn_tflite.CnnTflite;
import com.example.safedrive.activities.driveralert.MainActivity;
import com.example.safedrive.sqlite.DatabaseHelper;
import java.util.HashMap;
import kotlin.Metadata;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Metadata(
   mv = {1, 1, 16},
   bv = {1, 0, 3},
   k = 1,
   d1 = {"\u0000,\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0002\b\u0002\u0018\u00002\u00020\u0001B\u0005¢\u0006\u0002\u0010\u0002J&\u0010\u0003\u001a\u0004\u0018\u00010\u00042\u0006\u0010\u0005\u001a\u00020\u00062\b\u0010\u0007\u001a\u0004\u0018\u00010\b2\b\u0010\t\u001a\u0004\u0018\u00010\nH\u0016J\u001a\u0010\u000b\u001a\u00020\f2\u0006\u0010\r\u001a\u00020\u00042\b\u0010\t\u001a\u0004\u0018\u00010\nH\u0016¨\u0006\u000e"},
   d2 = {"Lcom/example/safedrive/nav_components/dashborad/Dashborad;", "Landroidx/fragment/app/Fragment;", "()V", "onCreateView", "Landroid/view/View;", "inflater", "Landroid/view/LayoutInflater;", "container", "Landroid/view/ViewGroup;", "savedInstanceState", "Landroid/os/Bundle;", "onViewCreated", "", "view", "app"}
)
public final class Dashborad extends Fragment {
   private HashMap _$_findViewCache;

   @Nullable
   public View onCreateView(@NotNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
      Intrinsics.checkParameterIsNotNull(inflater, "inflater");
      new DatabaseHelper((Context)this.getActivity());
      View v = inflater.inflate(1300142, container, false);
      String[] var10000 = new String[]{"safe_drive", "texting", "reaching_behind", "drinking", "operating_the_music", "talking_ont_the_phone"};
      return v;
   }

   public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
      Intrinsics.checkParameterIsNotNull(view, "view");
      ((ImageView)this._$_findCachedViewById(id.view_data)).setOnClickListener((OnClickListener)(new OnClickListener() {
         public final void onClick(View it) {
            Intent intent = new Intent((Context)Dashborad.this.getActivity(), ViewData.class);
            Dashborad.this.startActivity(intent);
         }
      }));
      ((ImageView)this._$_findCachedViewById(id.posenet)).setOnClickListener((OnClickListener)(new OnClickListener() {
         public final void onClick(View it) {
            Intent intent = new Intent((Context)Dashborad.this.getActivity(), Prediction.class);
            Dashborad.this.startActivity(intent);
         }
      }));
      ((ImageView)this._$_findCachedViewById(id.driveralert)).setOnClickListener((OnClickListener)(new OnClickListener() {
         public final void onClick(View it) {
            Intent intent = new Intent((Context)Dashborad.this.getActivity(), MainActivity.class);
            Dashborad.this.startActivity(intent);
         }
      }));
      ((ImageView)this._$_findCachedViewById(id.cnn)).setOnClickListener((OnClickListener)(new OnClickListener() {
         public final void onClick(View it) {
            Intent intent = new Intent((Context)Dashborad.this.getActivity(), CnnTflite.class);
            Dashborad.this.startActivity(intent);
         }
      }));
      super.onViewCreated(view, savedInstanceState);
   }

   public View _$_findCachedViewById(int var1) {
      if (this._$_findViewCache == null) {
         this._$_findViewCache = new HashMap();
      }

      View var2 = (View)this._$_findViewCache.get(var1);
      if (var2 == null) {
         View var10000 = this.getView();
         if (var10000 == null) {
            return null;
         }

         var2 = var10000.findViewById(var1);
         this._$_findViewCache.put(var1, var2);
      }

      return var2;
   }

   public void _$_clearFindViewByIdCache() {
      if (this._$_findViewCache != null) {
         this._$_findViewCache.clear();
      }

   }

   // $FF: synthetic method
   public void onDestroyView() {
      super.onDestroyView();
      this._$_clearFindViewByIdCache();
   }
}
// DashboradKt.java
package com.example.safedrive.nav_components.dashborad;

import kotlin.Metadata;

@Metadata(
   mv = {1, 1, 16},
   bv = {1, 0, 3},
   k = 2,
   d1 = {"\u0000\n\n\u0000\n\u0002\u0010\u000e\n\u0002\b\u0002\"\u000e\u0010\u0000\u001a\u00020\u0001X\u0082T¢\u0006\u0002\n\u0000\"\u000e\u0010\u0002\u001a\u00020\u0001X\u0082T¢\u0006\u0002\n\u0000¨\u0006\u0003"},
   d2 = {"ARG_PARAM1", "", "ARG_PARAM2", "app"}
)
public final class DashboradKt {
   private static final String ARG_PARAM1 = "param1";
   private static final String ARG_PARAM2 = "param2";
}

File: app/src/main/java/com/example/safedrive/nav_components/dashborad/Dashborad.kt
package com.example.safedrive.nav_components.dashborad

import android.content.ComponentName
import android.content.Intent
import android.os.Bundle
import android.view.*
import androidx.fragment.app.Fragment
import com.example.safedrive.R
import com.example.safedrive.activities.AppGuide
import com.example.safedrive.activities.Prediction
import com.example.safedrive.activities.ViewData
import com.example.safedrive.activities.cnn_tflite.CnnTflite
import com.example.safedrive.activities.driveralert.FaceTrackerActivity
import com.example.safedrive.activities.driveralert.MainActivity
import com.example.safedrive.sqlite.DatabaseHelper
import kotlinx.android.synthetic.main.dashborad.*


// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 * Use the [Dashborad.newInstance] factory method to
 * create an instance of this fragment.
 */
class Dashborad : Fragment() {


    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val myDb = DatabaseHelper(this.activity)

        val v = inflater.inflate(R.layout.dashborad, container, false)
        val values = arrayOf(
            "safe_drive",
            "texting",
            "reaching_behind",
            "drinking",
            "operating_the_music",
            "talking_ont_the_phone"
        )
//        val spinner = v.findViewById(R.id.spinner) as Spinner
//        val adapter = ArrayAdapter(
//            this.activity!!,
//            android.R.layout.simple_spinner_item,
//            values
//        )
//        adapter.setDropDownViewResource(android.R.layout.simple_dropdown_item_1line)
//        spinner.adapter = adapter
//        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
//            override fun onItemSelected(
//                parent: AdapterView<*>,
//                view: View?,
//                position: Int,
//                l: Long
//            ) {
//                when (position) {
//                    0 -> {
//                        textView.text="safe_drive"
//                    }
//                    1 -> {
//                        textView.text="texting"
//                    }
//                    2 -> {
//                        textView.text="reaching_behind"
//                    }
//                    3->{
//                        textView.text="drinking"
//                    }
//                    4->{
//                        textView.text="operating_the_music"
//
//                    }
//                    5->{
//                        textView.text="talking_ont_the_phone"
//                    }
//                }
//            }
//
//            override fun onNothingSelected(adapterView: AdapterView<*>?) {}
//        }


        return v
    }


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        view_data.setOnClickListener { val intent = Intent(activity, ViewData::class.java)
            startActivity(intent)
        }

        posenet.setOnClickListener { val intent = Intent(activity, Prediction::class.java)
            startActivity(intent)
        }
        driveralert.setOnClickListener { val intent = Intent(activity, MainActivity::class.java)
            startActivity(intent)
        }

//        cnn.setOnClickListener { val intent = Intent(activity, CnnTflite::class.java)
//            startActivity(intent)
//        }
        cnn.setOnClickListener {
            val intent =
                Intent(Intent.ACTION_MAIN)
            intent.component = ComponentName(
                "org.tensorflow.demo",
                "org.tensorflow.demo.ClassifierActivity"
            )
            startActivity(intent)


        }
        help.setOnClickListener{
            val intent=Intent(activity,AppGuide::class.java)
            startActivity(intent)
        }

        super.onViewCreated(view, savedInstanceState)
    }

}
